src/NextGen/batchManager.ts

```diff
@@ -0,0 +1,52 @@
+import {
+    Batch,
+    BatchEnvironmentEnum,
+    BaseEvent,
+} from '@mparticle/event-models';
+import { BaseMPMessage } from './core/baseMPMessage';
+import { isEmpty } from '../utils';
+
+// TODO: Sync this with existing API Client(s)
+export interface IMpApiClientImplementation {
+    uploadBatch(batch: Batch): void;
+}
+
+export default class BatchManager {
+    constructor(private api: IMpApiClientImplementation) {
+        this.api = api;
+    }
+
+    // QUESTION: Should this take an event or message as a param?
+    // Making an assumption that creating a batch should require a message
+    public createBatch(events: BaseMPMessage[]): Batch {
```

Comment:
I mainly bring up the question because in the android spec, `createBatch` does not take any params and I think that's probably because the spec was a rough draft.

On web, we actually take an [array of events and a user object](https://github.com/mParticle/mparticle-web-sdk/blob/master/src/batchUploader.ts#L199) so that we can order the batch's events based on session, so that's another process we should standardize on.

Reply:
@alexs-mparticle That's it. It was an outline to show interactions only. We can determine necessary params/dependencies